web/client/components/shapefile/ShapefileUploadAndStyle.jsx

```diff
@@ -127,14 +140,29 @@ const ShapeFileUploadAndStyle = React.createClass({
             this.props.onShapeError(e.message || e);
         });
     },
-    addToMap() {
+        addToMap() {
         this.props.shapeLoading(true);
         let styledLayer = this.props.selected;
         if (!this.state.useDefaultStyle) {
             styledLayer = StyleUtils.toVectorStyle(styledLayer, this.props.shapeStyle);
         }
+        let bbox = this.props.bbox;
         Promise.resolve(this.props.addShapeLayer( styledLayer )).then(() => {
             this.props.shapeLoading(false);
+
+            // calculate the max bbox that contain all shapefiles
+            let layers = this.props.layers;
+            layers[0].features.reduce((total, feature) => {
```

Comment:
Is it correct like this?
const bbox = layers[0].features.reduce((bboxtotal, feature) => {
                 let actualBBox = feature.geometry.bbox;
                 bboxtotal[0] = Math.min(bboxtotal[0], actualBBox[0]); // minx 
                 ....
                 bboxtotal[3] = Math.max(bboxtotal[3], actualBBox[3]); // maxy
}, this.props.bbox);